### PR TITLE
Support and test against the latest refactor-nrepl, cider-nrepl

### PR DESCRIPTION
**Pull request by vemv**
## Brief

There's been a refactor-nrepl upgrade, which:

* isn't a common occurrence
* revamped its internals, which makes it advisable for us to expand our CI matrix.

In the new matrix step, I also upgraded cider-nrepl, which uncovered the following issue: https://github.com/clojure-emacs/cider-nrepl/pull/668

This PR solves that issue as well.

## QA plan

* We have good test coverage for refactor-nrepl `clean-ns`
* Still, it'd be interesting if you tried this recent refactor-nrepl + cider-nrepl in Cursive and check if 'clean unused requires' keeps working.

## Author checklist

* [x] I have QAed the functionality
* [x] The PR has a reasonably reviewable size and a meaningful commit history
* [x] I have run the [branch formatter](https://github.com/nedap/formatting-stack/blob/332a419034ab46fad526a5592f4257353bd695b6/src/formatting_stack/branch_formatter.clj) and observed no new/significative warnings
* [x] The build passes
* [x] I have self-reviewed the PR prior to assignment
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [x] Correctness
  * [x] Robustness (red paths, failure handling etc)
  * [x] Modular design
  * [x] Test coverage
  * [x] Spec coverage
  * [x] Documentation
  * [x] Security
  * [x] Performance
  * [x] Breaking API changes
  * [x] Cross-compatibility (Clojure/ClojureScript)

## Reviewer checklist

* [x] I have checked out this branch and reviewed it locally, running it
* [x] I have QAed the functionality
* [x] I have reviewed the PR
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [x] Correctness
  * [ ] Robustness (red paths, failure handling etc)
  * [ ] Modular design
  * [x] Test coverage
  * [x] Spec coverage
  * [ ] Documentation
  * [x] Security
  * [x] Performance
  * [ ] Breaking API changes
  * [x] Cross-compatibility (Clojure/ClojureScript)

### CODE REVIEW

File: src/formatting_stack/project_parsing.clj [hunk 1]
```diff
@@ -22,12 +23,21 @@
        (map (speced/fn [^File f]
               (-> f .getCanonicalFile)))))
 
+(speced/defn ^{::speced/spec (spec/coll-of any? :min-count 1)} classpath-directories
```
Comment (vemv) [inline src/formatting_stack/project_parsing.clj]: One-off QAing for this defn:

```clj
(let [a (-> (classpath-directories)
            (find-files find/clj)
            (set))
      b (-> (classpath/classpath-directories)
            (find-files find/clj)
            (set))]
  (check! empty? (set/difference b a)))
```

(I considered and discarded adding this piece of logic to the test suite... it wouldn't pass in JDK11, and we don't necessarily have to test against 3rd party bugs anyway)

File: test/unit/formatting_stack/processors/test_runner/impl.clj [hunk 2]
```diff
@@ -22,8 +23,11 @@
 
 (deftest sut-consumers
   (let [corpus (project-namespaces)]
-    (are [input expected] (=  expected
-                              (sut/sut-consumers corpus input))
+
+    (check! (partial < 100) (-> corpus count))
```
Comment (thumbnail) [inline test/unit/formatting_stack/processors/test_runner/impl.clj]: `100` seems arbitrary,  what are you asserting?

**Comment by thumbnail** [comment]
Noticed your target-branch is `deleted`

File: test/unit/formatting_stack/processors/test_runner/impl.clj (hunk 2 above)
Comment (vemv) [reply test/unit/formatting_stack/processors/test_runner/impl.clj]: > 100 seems arbitrary, what are you asserting?

If it's < 100, something is wrong and the tests will fail rather opaquely

**Pull request merged**